weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts

```diff
@@ -466,14 +485,69 @@ const fetchEvaluationComparisonResults = async (
   const evalTraceIds = Object.values(summaryData.evaluationCalls).map(
     call => call.traceId
   );
+
   // First, get all the children of the evaluations (predictAndScoreCalls + summary)
   const evalTraceResProm = traceServerClient
     .callsStreamQuery({
       project_id: projectId,
       filter: {trace_ids: evalTraceIds, parent_ids: evaluationCallIds},
     })
     .then(predictAndScoreCallRes => {
-      // Then, get all the children of those calls (predictions + scores)
+      // Check for imperative evaluations and handle specially
+      const hasImperativeEvals = predictAndScoreCallRes.calls.some(
+        call =>
+          call.op_name.includes('Evaluation.predict_and_score:') ||
+          call.op_name.includes('Evaluation.summarize:')
+      );
+
+      // For imperative evaluations, we need to fetch all predict_and_score calls
+      // that might have a different op name pattern
+      if (hasImperativeEvals) {
+        // Add additional query to find imperative predict_and_score calls
+        return traceServerClient
+          .callsStreamQuery({
+            project_id: projectId,
+            filter: {
+              trace_ids: evalTraceIds,
+              op_names: ['Evaluation.predict_and_score:'],
+            },
+          })
+          .then(imperativePredictAndScoreCalls => {
+            // Combine with the original calls, removing duplicates
+            const combinedCalls = [
```

Comment:
maybe move this into a handler and split it into two functions. One which takes the combinedCalls result and one which takes the imperativePredictAndScoreCalls